src/main/java/org/graylog/integrations/aws/resources/AWSResource.java

```diff
@@ -100,35 +100,26 @@ public KinesisPermissionsResponse getPermissions() {
         return awsService.getPermissions();
     }
 
-    /**
-     * Get all available AWS CloudWatch log groups names for the specified region.
-     */
     @POST
     @Timed
     @Path("/cloudwatch/log_groups")
     @ApiOperation(value = "Get all available AWS CloudWatch log groups names for the specified region.")
     @RequiresPermissions(AWSPermissions.AWS_READ)
     @NoAuditEvent("This does not change any data")
-    public LogGroupsResponse getLogGroupNames(@ApiParam(name = "JSON body", required = true) @Valid @NotNull AWSRequestImpl awsRequest) {
-        return cloudWatchService.getLogGroupNames(awsRequest.region(), awsRequest.awsAccessKeyId(), awsRequest.awsSecretAccessKey());
+    public LogGroupsResponse getLogGroupNames(@ApiParam(name = "JSON body", required = true) @Valid @NotNull AWSRequestImpl request) {
+        return cloudWatchService.getLogGroupNames(request);
```

Comment:
@danotorrey is this a new method , that is able to extract all the information from the requeust ?